TFNativeOp.py

```diff
@@ -243,7 +243,12 @@ def make_dim_str(c):
     for in_idx, v in enumerate(in_info):
       out_idx = v.get("want_inplace", -1)
       if out_idx >= 0:
-        code_forward_io += "context->forward_ref_input_to_ref_output(%i, %i);\n" % (in_idx, out_idx)
+        code_forward_io += """
+          // Note: forward_ref_input_to_ref_output expects the input tensor to be a reference, and not a value;
+          //   otherwise it will trigger an assertion.
+          if (IsRefType(context->input_dtype({in_idx})))
+            context->forward_ref_input_to_ref_output({in_idx}, {out_idx});
```

Comment:
Now you removed the `else` part. Is that right?

Reply:
Yep - otherwise, TF quits with error and complains about more than one allocation on the output tensor. With the current state, all the tests seem to pass.

Reply:
Ah ok I see. Still not sure if that `forward_ref_input_to_ref_output` makes sense actually... But anyway, maybe we can leave it like that for now.